packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,104 +197,248 @@ class CupertinoAlertDialog extends StatelessWidget {
+// minimum button height at the bottom.
+//
+// After the content section is laid out, the action button section is allowed
+// to take up any remaining space that was not consumed by the content section.
+class _CupertinoDialogRenderBox extends RenderBox
+    with ContainerRenderObjectMixin<RenderBox, MultiChildLayoutParentData>,
+    RenderBoxContainerDefaultsMixin<RenderBox, MultiChildLayoutParentData> {
+  _CupertinoDialogRenderBox({
+    List<RenderBox> children,
+    bool isStacked = false,
+  }) : _isStacked = isStacked {
+    addAll(children);
+  }
 
-  /// The widget below this widget in the tree.
-  ///
-  /// Typically a [Text] widget.
-  final Widget child;
+  final bool _isStacked;
 
-  /// Whether the button is enabled or disabled. Buttons are disabled by
-  /// default. To enable a button, set its [onPressed] property to a non-null
-  /// value.
-  bool get enabled => onPressed != null;
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! MultiChildLayoutParentData)
+      child.parentData = new MultiChildLayoutParentData();
+  }
 
   @override
-  Widget build(BuildContext context) {
-    TextStyle style = _kCupertinoDialogActionStyle;
+  double computeMinIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
 
-    if (isDefaultAction) {
-      style = style.copyWith(fontWeight: FontWeight.w600);
-    }
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    return _kCupertinoDialogWidth;
+  }
 
-    if (isDestructiveAction) {
-      style = style.copyWith(color: CupertinoColors.destructiveRed);
-    }
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    return _computeIntrinsicHeight(width);
+  }
 
-    if (!enabled) {
-      style = style.copyWith(color: style.color.withOpacity(0.5));
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return _computeIntrinsicHeight(width);
+  }
+
+  double _computeIntrinsicHeight(double width) {
+    // Obtain references to the specific children we need lay out.
+    final _DialogChildren dialogChildren = _findDialogChildren();
+    final RenderBox content = dialogChildren.content;
+    final RenderBox actions = dialogChildren.actions;
+
+    final double contentHeight = content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = actions.getMaxIntrinsicHeight(width);
+    final double height = contentHeight + actionsHeight;
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  void performLayout() {
+    // Obtain references to the specific children we need lay out.
+    final _DialogChildren dialogChildren = _findDialogChildren();
+    final RenderBox content = dialogChildren.content;
+    final RenderBox actions = dialogChildren.actions;
+
+    final double minActionsHeight = actions.getMinIntrinsicHeight(constraints.maxWidth);
+
+    final Size maxDialogSize = constraints.biggest;
```

Comment:
what if the incoming constraints are unbounded?

Reply:
That's a good question.  What would that mean?

I don't think the height is a concern because maxDialogSize is only used to reference its width.

But what should we do with an unbounded width?  We could constrain it to the standard dialog width (which, functionally speaking, is already happening because the thing that uses this RenderBox is already imposing those constraints).  But I don't know if that's a great approach or not.

Any preferences?